packages/web/src/common/store/pages/history/lineups/sagas.ts

```diff
@@ -1,44 +1,54 @@
-import { LineupEntry, Track, UserTrackMetadata } from '@audius/common/models'
+import { LineupEntry, Track, UserTrack, UserTrackMetadata } from '@audius/common/models'
 import {
   accountSelectors,
   getContext,
   historyPageTracksLineupActions as tracksActions
 } from '@audius/common/store'
-import { removeNullable } from '@audius/common/utils'
+import { decodeHashId, encodeHashId, removeNullable } from '@audius/common/utils'
 import { keyBy } from 'lodash'
 import { call, select } from 'typed-redux-saga'
 
 import { processAndCacheTracks } from 'common/store/cache/tracks/utils'
 import { LineupSagas } from 'common/store/lineup/sagas'
 import { waitForRead } from 'utils/sagaHelpers'
+import { APIActivityV2, responseAdapter } from '@audius/common/services'
 const { getUserId } = accountSelectors
 const { prefix: PREFIX } = tracksActions
 
 function* getHistoryTracks() {
   yield* waitForRead()
 
-  const apiClient = yield* getContext('apiClient')
+  const audiusSdk = yield* getContext('audiusSdk')
+  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+  const sdk = yield* call(audiusSdk)
   try {
     const currentUserId = yield* select(getUserId)
     if (!currentUserId) return []
+    
+    const { data, signature } = yield* call([
+      audiusBackendInstance,
+      audiusBackendInstance.signDiscoveryNodeRequest
+    ])
+    const activity = yield* call(
+      [sdk.full.users, sdk.full.users.getUsersTrackHistory],
+      {
+        id: encodeHashId(currentUserId),
+        encodedDataMessage: data,
+        encodedDataSignature: signature,
+        limit: 100
+      }
+    )
+    const activityData = activity.data as APIActivityV2[]
+    const tracks = activityData.map(responseAdapter.makeActivity)
+      .filter(removeNullable) as UserTrackMetadata[]
 
-    const activity = yield* call([apiClient, apiClient.getUserTrackHistory], {
-      currentUserId,
-      userId: currentUserId,
-      limit: 100
-    })
-
-    const activityTracks: UserTrackMetadata[] = activity
-      .map((a) => a.track)
-      .filter(removeNullable)
-
-    const processedTracks = yield* call(processAndCacheTracks, activityTracks)
+    const processedTracks = yield* call(processAndCacheTracks, tracks)
     const processedTracksMap = keyBy(processedTracks, 'track_id')
 
     const lineupTracks: Track[] = []
-    activity.forEach((activity) => {
-      const trackMetadata = activity.track
-        ? processedTracksMap[activity.track.track_id]
+    activityData.forEach((activity) => {
+      const trackMetadata = activity.item
+        ? processedTracksMap[decodeHashId(activity.item.id)!]
```

Comment:
should this be `[decodeHashId(activity.item.id!)]` (non null assertion of id not the decode output)

Reply:
decodeHashId actually can return null, so this is right. It should never happen though, unless the provided input is a strange input like `NaN`